### ISSUE TITLE
SwiftTask 2.6.1 prevents ReactKit from building when using Carthage

### ISSUE THREAD

**Issue by isair** [ReactKit/ReactKit#16]
So far I have tried using ReactKit 0.7.0, 0.6.1 and 0.6.0. They all failed to build with an error because of SwiftTask when using the command `carthage update`. The error comes down to these two lines in my build log:
```
/Users/UserName/Projects/ProjectName/Carthage/Checkouts/ReactKit/Carthage/Checkouts/SwiftTask/SwiftTask/SwiftTask.swift:758:5: error: use of unresolved identifier 'objc_sync_enter'
    objc_sync_enter(object)
    ^
/Users/UserName/Projects/ProjectName/Carthage/Checkouts/ReactKit/Carthage/Checkouts/SwiftTask/SwiftTask/SwiftTask.swift:760:5: error: use of unresolved identifier 'objc_sync_exit'
    objc_sync_exit(object)
    ^
```

**Comment by inamiy** [comment]
Thank you for feedback.
But I couldn't reproduce the bug you mentioned.

What version do you use for Xcode/Swift/Carthage?
Can you try building demo app e.g. https://github.com/ReactKit/ReactKitCalculator
to reproduce same bug?

**Comment by isair** [comment]
Updated to the latest Carthage (0.6.2) but it still fails due to the same errors. My Xcode version is 6.1.1 (6A2008a). Tested running `carthage update` inside ReactKitCalculator, sadly the same problem.
![screenshot 2015-02-25 16 11 41](https://cloud.githubusercontent.com/assets/988213/6383764/0508c36e-bd09-11e4-86ca-902e5ec3f3d6.png)

**Comment by inamiy** [comment]
Can you try `xcodebuild -workspace SwiftTask.xcworkspace -scheme SwiftTask-iOS -configuration Release` in `Carthage/Checkouts/SwiftTask` directory and see if same issue happens?

If so, there might be some Xcode's old build caches bothering.
Please clean up DerivedData (`~/Library/Developer/Xcode/DerivedData`) and try once more.

**Comment by LightD** [comment]
Hi @inamiy 
I had the same exact problem when I was trying to embed ReactKit Using Carthage.. I'm using swift 1.2, but that's not the issue (coz I modified the files and if I open the workspace on it's own it build perfectly).. but I just can't build ReactKit using Carthage.. what happens after I run your command which you mentioned in the previous comment? how do I build the "manually modified" projects? :\ 
or could it be because I'm using the beta version of xcode and carthage build is screwing things around :\

**Comment by inamiy** [comment]
@LightD
Thanks for another report.
Though I haven’t encountered this bug yet, it seems reproducible, so I will investigate this issue more.

If you could successfully build both SwiftTask and ReactKit projects inside `Carthage/Checkout` folder,
then there seems to have no issues for `xcodebuild`-ing each frameworks,
so `carthage build` should just work too and you'll see the built frameworks inside `Carthage/Build` folder.

As I said earlier, can you try cleaning up `DerivedData` folder and `carthage build` again?

P.S. 
For Swift 1.2, I tried `xcode-select -switch` to Xcode 6.3 beta, `carthage build`, and embed to test app,
and the all just worked fine...

```
# Cartfile for Swift 1.2
github "ReactKit/ReactKit" "swift/1.2"
github "ReactKit/SwiftTask" "swift/1.2"
```

**Comment by LightD** [comment]
Actually.. this could've been a noob mistake.. sorry :stuck_out_tongue_closed_eyes: 
So.. I indeed had to open the "checked out" project and modify the files for swift 1.2 modifications (coz 1. I didn't know I could specify branch name from carthage.. 2. from what I've noticed, the swift branches are not updated to the latest master code, so went with modifying them).
my mistake is that I was always doing `carthage update` and I wasn't doing `carthage build`.
